tests/admin/views/test_releases_v2.py

```diff
@@ -0,0 +1,1293 @@
+        dbo.release_assets.t.select()
+        .where(dbo.release_assets.name == "Firefox-60.0b3-build1")
+        .where(dbo.release_assets.path.in_((".platforms.Linux_x86_64-gcc3.locales.de",)))
+        .execute()
+        .fetchall()
+    )
+    assert not removed_locales
+    got_newde = (
+        dbo.release_assets.t.select()
+        .where(dbo.release_assets.name == "Firefox-60.0b3-build1")
+        .where(dbo.release_assets.path == ".platforms.Linux_x86_64-gcc3.locales.newde")
+        .execute()
+        .fetchone()
+        .data
+    )
+    assert newde == got_newde
+    base_history = get_release_history("Firefox-60.0b3-build1")
+    assert len(base_history) == 2
+    de_history = get_release_assets_history("Firefox-60.0b3-build1", ".platforms.Linux_x86_64-gcc3.locales.de")
+    assert len(de_history) == 3
+    # Check to make sure "changed_by" was set
+    assert "-bob.json" in de_history[2][0]
+    af_history = get_release_assets_history("Firefox-60.0b3-build1", ".platforms.Linux_x86_64-gcc3.locales.af")
+    assert len(af_history) == 3
+    assert "-bob.json" in af_history[2][0]
+    newde_history = get_release_assets_history("Firefox-60.0b3-build1", ".platforms.Linux_x86_64-gcc3.locales.newde")
+    assert len(newde_history) == 2
+    assert "-bob.json" in newde_history[1][0]
+
+    unchanged_locale = (
+        dbo.release_assets.t.select()
+        .where(dbo.release_assets.name == "Firefox-60.0b3-build1")
+        .where(dbo.release_assets.path == ".platforms.Linux_x86_64-gcc3.locales.zh-TW")
+        .execute()
+        .fetchone()
+    )
+    assert unchanged_locale["data_version"] == 1
+    unchanged_locale_history = get_release_assets_history("Firefox-60.0b3-build1", ".platforms.Linux_x86_64-gcc3.locales.zh-TW")
+    assert len(unchanged_locale_history) == 2
+
+
+@pytest.mark.usefixtures("releases_db", "mock_verified_userinfo")
+def test_put_succeeds_for_nonsplit_release(api, cdm_17):
+    cdm_17 = deepcopy(cdm_17)
+    cdm_17["vendors"]["gmp-eme-adobe"]["platforms"]["WINNT_x86-msvc"]["filesize"] = 5555555555
+
+    ret = api.put("/v2/releases/CDM-17", json={"blob": cdm_17, "product": "CDM", "old_data_versions": versions_dict()})
+    assert ret.status_code == 200, ret.data
+    assert ret.json == {".": 2}
+
+    base_blob = dbo.releases_json.t.select().where(dbo.releases_json.name == "CDM-17").execute().fetchone().data
+    assert base_blob == cdm_17
+    history = get_release_history("CDM-17")
+    assert len(history) == 3
+    # Check to make sure "changed_by" was set
+    assert "-bob.json" in history[2][0]
+
+
+@pytest.mark.usefixtures("releases_db", "mock_verified_userinfo")
+def test_post_fails_when_release_doesnt_exist(api):
+    blob = {"detailsUrl": "https://newurl", "platforms": {"Darwin_x86_64-gcc3-u-i386-x86_64": {"locales": {"de": {"buildID": "999999999999999"}}}}}
+    old_data_versions = versions_dict()
+    assert old_data_versions["platforms"]["Darwin_x86_64-gcc3-u-i386-x86_64"]["locales"]["de"]
```

Comment:
`populate_versions_dict(blob)` would be option here, given the test is about getting an error before trying to modify a release ?

Reply:
Could do, though that might be a less realistic request?

Reply:
Perhaps I'm missing something, but `old_data_versions = populate_versions_dict(blob)` seems to produce the same dict as the two lines here.

Reply:
On reading other tests (like a tweaked `test_put_add_scheduled_change_base_and_locale`) I see that the suggestion would be sending unnecessary data. Where`populate_versions_dict` is used it's for big blobs, although with a similar amount of change as this test. So I guess I don't feel hugely strongly any more, but something something consistency ?

Reply:
> Perhaps I'm missing something, but old_data_versions = populate_versions_dict(blob) seems to produce the same dict as the two lines here.

With fresh eyes, I think you're right about this. My brain was shortcutting `populate_versions_dict` as "always adds many platforms and locales". We may as well use `populate_versions_dict` for consistency, as you suggest.